File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/2872/Protocolo.java
import java.util.*;

public class Protocolo{
    private static void bubbleSort(String[] array, int n){
        for(int i = 0; i < n - 1; i++){
            for(int j = 0; j < n - 1 - i; j++){
                if(array[j].compareTo(array[j + 1]) > 0){
                    String temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }
    
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        while(sc.hasNextLine()){
            String line = sc.nextLine();

            if(line.equals("1")){
                String[] pack = new String[10];
                int count = 0; 

                while(sc.hasNextLine()){
                    String line1 = sc.nextLine();

                    if(line1.equals("0")){
                        break;
                    }

                    pack[count++] = line1;
                }

                bubbleSort(pack, count);

                for(int i = 0; i < count; i++){
                    System.out.println(pack[i]);
                }

                System.out.println();
            }
        }

        sc.close();
    }
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/2872/Protocolo.c
#include <stdio.h>
#include <string.h>

void bubbleSort(char arr[][51], int n){
    for(int i = 0; i < n - 1; i++){
        for(int j = 0; j < n - 1 - i; j++){
            if(strcmp(arr[j], arr[j + 1]) > 0){
                char temp[51];
                strcpy(temp, arr[j]);
                strcpy(arr[j], arr[j + 1]);
                strcpy(arr[j + 1], temp);
            }
        }
    }
}

int main(){
    char line[51];

    while(fgets(line, sizeof(line), stdin) != NULL){
        if(line[strlen(line) - 1] == '\n')
            line[strlen(line) - 1] = '\0';

        if(strcmp(line, "1") == 0){
            char pack[10][51];
            int count = 0;

            while(fgets(line, sizeof(line), stdin) != NULL){
                if(line[strlen(line) - 1] == '\n')
                    line[strlen(line) - 1] = '\0';

                if(strcmp(line, "0") == 0){
                    break;
                }

                strcpy(pack[count++], line);
            }

            bubbleSort(pack, count);

            for(int i = 0; i < count; i++){
                printf("%s\n", pack[i]);
            }

            printf("\n");
        }
    }

    return 0;
}
